eip-0030.md

```diff
@@ -0,0 +1,329 @@
+
+2. **Liquidify Pool**: The reverse swap, selling of Dexy tokens, is done via a Liquidity Pool (LP) which also permits buying Dexy tokens. The LP
+   primarily uses the logic of Uniswap V2. The difference is that the LP also takes as input the oracle pool rate and uses that to modify certain logic. In particular,
+   redeeming of LP tokens is not allowed when the oracle pool rate is below a certain percent (say 90%) of the LP rate.
+
+3. In case the oracle pool rate is higher than LP rate, then traders can do arbitrage by minting Dexy tokens from the emission box and
+   selling them to the LP.
+
+4. In case the oracle pool rate is lower than LP rate, then the Ergs collected in the emission box can be used to bring the rate back up by performing a swap.
+   We call this the "top-up swap".
+
+The swap logic is encoded in a **swapping** contract.
+
+There is another contract, the **tracking** contract that is responsible for tracking the LP's state. In particular, this contract
+tracks the block at which the "top-up-swap" is initiated. The swap can be initiated when the LP rate falls below 90%.
+Once initiated, if the LP rate remains below the oracle pool rate for a certain threshold number of blocks, the swap can be compleded.
+On the other hand, if before the threshold the rate goes higher than oracle pool then the swap must be aborted.
+
+The LP uses a "cross-counter" to keep count of the number of times the LP rate has crossed the oracle pool rate (from below or above) in a swap transaction.
+If the cross-counter is preserved at swap initiation and completion then swap is valid, else it is aborted. This logic is present in the swapping box.
+
+## Emission Contract
+
+```scala
+{ 
+  // This box: (dexyUSD emission box)
+  //   tokens(0): emissionNFT identifying the box
+  //   tokens(1): dexyUSD tokens to be emitted
+  
+  val selfOutIndex = getVar[Int](0).get
+  
+  val oraclePoolNFT = fromBase64("RytLYlBlU2hWbVlxM3Q2dzl6JEMmRilKQE1jUWZUalc=") // to identify oracle pool box
+  val swappingNFT = fromBase64("Fho6UlBlU2hWbVlxM3Q2dzl6JEMmRilKQE1jUWZUalc=") // to identify swapping box for future use
+  
+  val validEmission = {
+    val oraclePoolBox = CONTEXT.dataInputs(0) // oracle-pool (v1 and v2) box containing rate in R4
+  
+    val validOP = oraclePoolBox.tokens(0)._1 == oraclePoolNFT
+  
+    val oraclePoolRate = oraclePoolBox.R4[Long].get // can assume always > 0 (ref oracle pool contracts) NanoErgs per USD
+  
+    val selfOut = OUTPUTS(selfOutIndex)
+  
+    val validSelfOut = selfOut.tokens(0) == SELF.tokens(0) && // emissionNFT and quantity preserved
+                     selfOut.propositionBytes == SELF.propositionBytes && // script preserved
+                     selfOut.tokens(1)._1 == SELF.tokens(1)._1 && // dexyUSD tokenId preserved
+                     selfOut.value > SELF.value // can only purchase dexyUSD, not sell it
```

Comment:
I believe the comment is misinterpreted. The full comment should read "users can only purchase dexyUSD". 

Maybe we can also write it as "box can only sell dexyUSD"?

Also since the contract sells dexyUSD, shouldn't selfOut.value be more than SELF.value?  (i.e., shouldn't Ergs increase?)